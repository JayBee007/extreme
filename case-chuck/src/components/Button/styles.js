/* eslint-disable import/prefer-default-export */
import color from "color";

export const btn = theme => ({
  btn: {
    fontSize: "1.6rem",
    padding: "1rem 1.5rem",
    color: theme.colors.white,
    backgroundColor: ({ color: clr }) => theme.colors[clr],
    borderColor: ({ color: clr }) => theme.colors[clr],
    fontWeight: "bold",
    borderRadius: "0.2rem",
    transition: "all .15s ease-in-out",
    "&:hover": {
      backgroundColor: ({ color: clr }) =>
        color(theme.colors[clr])
          .darken(0.2)
          .hex(),
      borderColor: ({ color: clr }) =>
        color(theme.colors[clr])
          .darken(0.2)
          .hex()
    }
  }
});
