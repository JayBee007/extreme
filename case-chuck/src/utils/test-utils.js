/* eslint-disable import/prefer-default-export */
import React from "react";

import ThemeProvider from "../theme/ThemeProvider";

export const applyTheme = component => {
  return <ThemeProvider>{component}</ThemeProvider>;
};
