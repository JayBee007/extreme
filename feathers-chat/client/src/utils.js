export const getAuthToken = () => {
  try {
    const token = window.localStorage.getItem('feathers-jwt');
    return token;
  } catch (error) {
    return null;
  }
}

export const removeAuthToken = () => {
  window.localStorage.clear();
}

export const isAuthenticated = () => (toState, fromState)=> {

  return new Promise((resolve,reject) => {
    const token = getAuthToken();
    
    if(token) {
      reject({
        redirect: { name: 'chat'}
      })
    }else {
      resolve();
    }

  })
};