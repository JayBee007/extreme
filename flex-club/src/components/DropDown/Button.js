import styled from 'styled-components';

const Button = styled.button`
  background-color: transparent;
  border: none;
  position: absolute;
  right: 0;
  top: 0;
  cursor: pointer;
  width: 4.5rem;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 100%;
  outline: none;
`;

export default Button;
