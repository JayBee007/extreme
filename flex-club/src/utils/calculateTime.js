const calculateTime = (utcDate, offset) => {
  const localDate = new Date(utcDate * 1000);
  const localTime = localDate.getTime();
  const localOffset = localDate.getTimezoneOffset() * 60000;
  const utcTime = localTime + localOffset;
  const targetOffset = offset / 60 / 60;

  // eslint-disable-next-line prettier/prettier
  const targetTime = utcTime + 3600000 * targetOffset;

  return new Date(targetTime);
};

export default calculateTime;
