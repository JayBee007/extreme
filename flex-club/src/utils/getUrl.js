const getUrl = (type, longitude, latitude) => {
  return `https://api.openweathermap.org/data/2.5/${type}?lat=${latitude}&lon=${longitude}&units=metric&appid=${process.env.REACT_APP_API_KEY}`;
};

export default getUrl;
