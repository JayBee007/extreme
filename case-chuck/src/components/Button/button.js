import React from "react";
import PropTypes from "prop-types";
import withStyle from "react-jss";

import { btn } from "./styles";

const Button = props => {
  const { children, classes, handleClick } = props;

  const clickHandle = () => {
    handleClick();
  };
  return (
    <button type="button" className={classes.btn} onClick={clickHandle}>
      {children}
    </button>
  );
};

Button.propTypes = {
  children: PropTypes.node.isRequired,
  classes: PropTypes.object.isRequired,
  handleClick: PropTypes.func.isRequired
};

export default withStyle(btn)(Button);
