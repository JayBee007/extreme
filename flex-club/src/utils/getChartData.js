import calculateTime from 'utils/calculateTime';

const getChartData = (weatherData = {}) => {
  const hashMap = {};
  const { list, city } = weatherData;
  let firstDay;
  let secondDay;
  if (!list) {
    return hashMap;
  }

  for (let i = 0; i < list.length; i += 1) {
    const currentTimeSegment = list[i];
    const dt = calculateTime(currentTimeSegment.dt, city.timezone);

    const currentIteratedDate = dt.getDate();
    const hours = dt.getHours();
    const minutes = dt.getMinutes();
    const computedMinutes =
      minutes.toString().length > 1 ? minutes : `0${minutes}`;

    if (!hashMap[currentIteratedDate]) {
      // Flag to see if this segment is today
      // Need it to later add the data if 24 hrs are not complete
      if (!firstDay) {
        firstDay = currentIteratedDate;
      } else if (firstDay && !secondDay) {
        secondDay = currentIteratedDate;
      }
      hashMap[currentIteratedDate] = {
        id: currentTimeSegment.dt,
        data: [
          {
            time: `${hours}:${computedMinutes}`,
            temp: currentTimeSegment.main.temp
          }
        ]
      };
    } else if (hashMap[currentIteratedDate]) {
      hashMap[currentIteratedDate] = {
        ...hashMap[currentIteratedDate],
        data: [
          ...hashMap[currentIteratedDate].data,
          {
            time: `${hours}:${computedMinutes}`,
            temp: currentTimeSegment.main.temp
          }
        ]
      };
    }
  }

  // Api returns data in breaks of three hours so for 24 hrs
  // 24 / 3 = 8, if we have eight elements in the data array
  // means we have calculated 24hrs worth of data if not
  // we copy the remaining from next day
  if (hashMap[firstDay].data.length < 8) {
    const remainingHours = 8 - hashMap[firstDay].data.length;
    const nextDayData = hashMap[secondDay].data;

    for (let i = 0; i < remainingHours; i += 1) {
      hashMap[firstDay].data = [...hashMap[firstDay].data, nextDayData[i]];
    }
  }
  return Object.values(hashMap);
};

export default getChartData;
