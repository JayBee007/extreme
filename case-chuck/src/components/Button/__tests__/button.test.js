import React from "react";
import renderer from "react-test-renderer";

import { applyTheme } from "../../../utils/test-utils";
import Button from "../Button";

describe("Button Component", () => {
  let button = null;
  const handleClick = jest.fn();
  beforeEach(() => {
    button = applyTheme(<Button handleClick={handleClick}>Button Text</Button>);
  });

  test("basic render", () => {
    const component = renderer.create(button);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("handleClick called with an event", () => {
    const component = renderer.create(button);
    const tree = component.toJSON();
    tree.props.onClick();
    expect(handleClick).toHaveBeenCalled();
  });
});
