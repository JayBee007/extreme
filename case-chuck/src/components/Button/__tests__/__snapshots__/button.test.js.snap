// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component basic render 1`] = `
<button
  className="Button-btn-0-1-1"
  onClick={[Function]}
  type="button"
>
  Button Text
</button>
`;
